File: elastic_drop_new.py
#!/usr/bin/python

import requests


x = 9
host = 'elk-102.ix.km'
exludes = 'filename1 filename2 filename3'

response = requests.get('http://%s:9200/_cluster/health' % host)
print response
if response.status_code != 200:
	# TODO: Do error message to log file
	print ('Wrong response, %s') % response.status_code
	exit (1)

print response.status_code

responseBody  = response.json()
if responseBody['status'] != 'green':
	# TODO: Do error message to log file
	print 'Status not green do nothing'
	exit (1)

print ('Cluster Status is: %s') % responseBody['status']

response = requests.get('http://%s:9200/_cat/indices' % host)
responseBody = response.text
#print responseBody
responseData = responseBody.split('\n')
for i in responseData:
	print i

# i = len(text)
#indices = []
# while x < i:
# indices.append(text[x])
# x = x + 8
# print indices
# print len(indices)
# llen = len(indices) - 60
# print llen
# if llen >= 9:
# indices.sort()
# print indices
# for elem in range(9,llen):
# print 'Delete: ', indices[elem]
#     r = requests.delete('http://elk-102.ix.km:9200/'+indices[elem])
#     print r

File: elastic_drop_new.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"log"
	"net/http"
)

const (
	elasticHost     = "elk-102.ix.km"
	clusterStatusOk = "green"
	excludeFile     = "./elastic_exlude.txt"
)

type myStruct struct {
	Status string `json:"status"`
}

type myConfig struct {
	ExcludedFileNames []string
}

var (
	cluster myStruct
	config  myConfig
)

func main() {
	bytes, err := ioutil.ReadFile(excludeFile)
	if err != nil {
		log.Fatalln(err)
	}
	err = json.Unmarshal(bytes, &config)
	if err != nil {
		log.Fatalf("Error: %s", err)
	}
	if len(config.ExcludedFileNames) == 0 {
		log.Fatalln("Exclude file is empty")
	}

	ClusterStatusUrl := fmt.Sprint("http://" + elasticHost + ":9200/_cluster/health?pretty")
	resp, err := http.Get(ClusterStatusUrl)
	if err != nil {
		log.Fatalf("Error: %s", err)
	}
	defer resp.Body.Close()

	if resp.StatusCode != http.StatusOK {
		log.Fatalf("Response status: %v", resp.Status)
	}

	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		log.Fatalf("Couldn't read data: %v", body)
	}
	err = json.Unmarshal(body, &cluster)
	if err != nil {
		log.Fatalf("error:", err)
	}
	if cluster.Status != clusterStatusOk {
		log.Fatalf("Cluster status is: %s!!! Do nothing ", cluster.Status)
	}

	ClusterIndicesUrl := fmt.Sprint("http://" + elasticHost + ":9200/_cat/indices?pretty")
	resp, err = http.Get(ClusterIndicesUrl)
	if err != nil {
		log.Fatalf("Error: %s", err)
	}
	defer resp.Body.Close()

	if resp.StatusCode != http.StatusOK {
		log.Fatalf("Response status: %v", resp.Status)
	}
	body, err = ioutil.ReadAll(resp.Body)
	if err != nil {
		log.Fatalf("Couldn't read data: %v", body)
	}
	fmt.Println(string(body))
}